### PR TITLE
Add Garden Plots Widget

### PR DESCRIPTION
**Pull request by viciscat**
Adds a nice little widget that shows your plots and highlights the ones with a pest problem.

https://github.com/user-attachments/assets/c95a3633-6a27-4424-a1d9-96219490b305

### CODE REVIEW

File: src/main/java/de/hysky/skyblocker/skyblock/garden/GardenPlotsWidget.java [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+	@Override
+	protected boolean isValidClickButton(int button) {
+		return super.isValidClickButton(button) && hoveredSlot != -1;
+	}
+
+	@Override
+	protected void appendClickableNarrations(NarrationMessageBuilder builder) {
+	}
+
+	private record GardenPlot(Item item, String name) {
```
Comment (kevinthegreat1) [inline src/main/java/de/hysky/skyblocker/skyblock/garden/GardenPlotsWidget.java:294]:
```suggestion
	// The name is just the plot number.
	private record GardenPlot(Item item, String name) {
```

**Review by kevinthegreat1 — CHANGES_REQUESTED** [review]
Some suggestions and I think you're good.

File: src/main/java/de/hysky/skyblocker/skyblock/garden/GardenPlotsWidget.java [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+			updateInfestedFromTab();
+		}
+	}
+
+	private void updateInfestedFromTab() {
+		infectedPlots.clear();
+		for (int i = 0; i < PlayerListMgr.getSize(); i++) {
+			Text displayName = PlayerListMgr.getRaw(i).getDisplayName();
+			if (displayName == null) continue;
+			String string = displayName.getString();
```
[the remark this answers is not in the document]
Comment (viciscat) [reply src/main/java/de/hysky/skyblocker/skyblock/garden/GardenPlotsWidget.java:262]: playerStringList trims the text. I use the space that hypixel adds in front for extra assurance that it's the correct line. But honestly i don't know where else "Plots: " would show up in TAB so why not

File: src/main/java/de/hysky/skyblocker/skyblock/garden/GardenPlotsWidget.java [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+	}
+
+	@Override
+	protected void renderWidget(DrawContext context, int mouseX, int mouseY, float delta) {
+		TextRenderer textRenderer = MinecraftClient.getInstance().textRenderer;
+		MatrixStack matrices = context.getMatrices();
+		matrices.push();
+		matrices.translate(getX(), getY(), 0);
+
+		context.drawTexture(BACKGROUND_TEXTURE, 0, 0, 0, 0, getWidth(), getHeight(), getWidth(), getHeight());
```
[the remark this answers is not in the document]
Comment (viciscat) [reply src/main/java/de/hysky/skyblocker/skyblock/garden/GardenPlotsWidget.java:193]: real. funnily enough, this texture exists and yet it's not nine slice https://mcasset.cloud/1.21/assets/minecraft/textures/gui/demo_background.png

File: src/main/java/de/hysky/skyblocker/skyblock/garden/GardenPlotsWidget.java (hunk 1 above)
Comment (viciscat) [reply src/main/java/de/hysky/skyblocker/skyblock/garden/GardenPlotsWidget.java:294]: it's not, you can rename your plots

Comment (kevinthegreat1) [reply src/main/java/de/hysky/skyblocker/skyblock/garden/GardenPlotsWidget.java:296]: Nevermind then.

**Review by kevinthegreat1 — DISMISSED** [review]

**Comment by kevinthegreat1** [comment]
This needs to be updated. To make it easier, you should use merge instead of rebase. (This also means that we'll squash the pr.)

**Review by kevinthegreat1 — DISMISSED** [review]

**Review by AzureAaron — APPROVED** [review]
looks good

**Review by AzureAaron — CHANGES_REQUESTED** [review]
Nvm this has a crash:
```
[14:11:16] [Render thread/ERROR]: Unreported exception thrown!
java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
	at java.base/java.util.ImmutableCollections$ListN.get(ImmutableCollections.java:687) ~[?:?]
	at knot/de.hysky.skyblocker.skyblock.garden.GardenPlotsWidget.updateInfestedFromTab(GardenPlotsWidget.java:263) ~[skyblocker-1.22.2-alpha-pr-929-3845601+1.21.3.jar:?]
	at knot/de.hysky.skyblocker.skyblock.garden.GardenPlotsWidget.<init>(GardenPlotsWidget.java:186) ~[skyblocker-1.22.2-alpha-pr-929-3845601+1.21.3.jar:?]
	at knot/net.minecraft.class_490.handler$cce000$skyblocker$addGardenPlotsWidget(class_490.java:1076) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_490.method_25426(class_490.java:51) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_437.method_25423(class_437.java:308) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_310.method_1507(class_310.java:1203) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_310.method_1508(class_310.java:2005) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_310.method_1574(class_310.java:1870) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_310.method_1523(class_310.java:1313) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_310.method_1514(class_310.java:933) [client-intermediary.jar:?]
	at knot/net.minecraft.client.main.Main.main(Main.java:265) [client-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480) [fabric-loader-0.16.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.16.9.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.16.9.jar:?]
```

**Comment by viciscat** [comment]
> Nvm this has a crash:

Should be impossible but sure lol, `playerList` and `playerStringList` in `PlayerListMgr` should be "synced" but yea idk 🤷

**Review by kevinthegreat1 — APPROVED** [review]

**Comment by x-yingcan-x** [comment]
missingno
do you forget to add texture?

**Comment by viciscat** [comment]
lmao yes

**Pull request merged**